### PR TITLE
custom bot names

### PR DESCRIPTION
**Pull request by mjkzy**

### CODE REVIEW

**Comment by DoktorSAS** [comment]
Whats is the json for? 
Couldn't it be done like iw4, iw5? That is, use the classic bots.txt file and each line represents a name.
I think using a text file like that is more user firendly

**Comment by mjkzy** [comment]
sure, but nothing wrong with using json? you're just making a array of names and i'm just using the json submodule we already use so this doesn't change much. i like json

**Comment by mjkzy** [comment]
uses `h1-mod/bots.txt` now, only side effect i've just now noticed is that when `/map_rotate` is ran and you spawn in and spawn a bot, SV_GetRandomBotName is called twice and the first name is ignored but the second one works. i assume this is because the first bot just doesn't spawn, but either way, it's not a huge issue. though, doing `/fast_restart` will use the first name and only call SV_GetRandomBotName once

**Comment by mjkzy** [comment]
if i do custom clan tags, back to json it is!

File: src/client/component/bots.cpp [hunk 1]
```diff
@@ -70,6 +71,58 @@ namespace bots
...
+		std::vector<std::string> bot_names{};
+
+		void load_bot_data()
+		{
+			static const char* bots_txt = "h1-mod/bots.txt";
+
+			std::string bots_content;
+			if (!utils::io::read_file(bots_txt, &bots_content))
+			{
+				console::error(utils::string::va("%s was not found.\n", bots_txt));
```
Comment (fedddddd) [inline src/client/component/bots.cpp:85]: No need to print this imo

File: src/client/component/bots.cpp [hunk 2]
```diff
@@ -70,6 +71,58 @@ namespace bots
...
+				return;
+			}
+
+			auto names = utils::string::split(bots_content, '\n');
+			for (auto& name : names)
+			{
+				name = utils::string::replace(name, "\r", "");
+#ifdef DEBUG
+				console::info(utils::string::va("Loading name '%s' for bot names...\n", name.data()));
```
Comment (fedddddd) [inline src/client/component/bots.cpp:94]: Same here

File: src/client/component/bots.cpp [hunk 3]
```diff
@@ -70,6 +71,58 @@ namespace bots
...
+
+		const char* get_random_bot_name()
+		{
+			if (bot_names.empty())
+			{
+				load_bot_data();
+			}
+
+#ifdef DEBUG
+			console::info(utils::string::va("Bot ID is %d\n", bot_id));
```
Comment (fedddddd) [inline src/client/component/bots.cpp:113]: Same here

File: src/client/component/bots.cpp [hunk 4]
```diff
@@ -70,6 +71,58 @@ namespace bots
...
+				console::info(utils::string::va("Loading name '%s' for bot names...\n", name.data()));
+#endif
+				if (!name.empty())
+				{
+					bot_names.emplace_back(name);
+				}
+			}
+		}
+
+		static size_t bot_id = 0;
```
Comment (fedddddd) [inline src/client/component/bots.cpp:103]: Remove the static, or move it inside get_random_bot_name

**Review by fedddddd — CHANGES_REQUESTED** [review]

File: src/client/component/bots.cpp (hunk 1 above)
Comment (mjkzy) [reply src/client/component/bots.cpp]: left from testing, fixed

**Pull request merged**